### PR TITLE
Refactor doc and templates

### PR DESCRIPTION
**Pull request by shashiranjan84**
- make SSL port optional
- additional Kong configuration
- make migrations optional
- Kong parameters detail added

### CODE REVIEW

**Comment by shashiranjan84** [comment]
@stackedsax would appreciate your feedback on this PR.

**Comment by stackedsax** [comment]
Looks good so far!

Some general questions:

1. Is it stating that obvious too much to tell people to `git clone` the repo first?  I feel like a few people might come expecting that `helm install` just magically pulls the chart from kubernetes/charts/stable.  I know it might feel wrong to be that specific, but can't hurt....
2. In the README, were you leaving the loadBalancerIP default values blank on purpose?  I think the default value for both of them is 'null', isn't it?
3. There are some really subtle grammatical things that I could fix in the morning, if you'd like.
4. Might it be nice to include a command to check that kong is running.  Something like:
`kubectl exec $(kubectl get pods -l app=kong-kong -o jsonpath='{.items[0].metadata.name}') -- curl -s http://kong-kong-admin:8001/status` 
But that example still requires people fill in their app name, so it may confuse people as much as it helps.  Eh, no big deal if something like this doesn't make it in this round.

But it's looking like a good and proper chart.  Nice work!

**Comment by stackedsax** [comment]
Do you want to list `cassandra.enabled` and `postgres.enabled`, too?

Since I got tripped up by the multiple flags to switch db's, I'd be happy to add a section explaining that nuance.

**Comment by shashiranjan84** [comment]
> Is it stating the obvious too much to tell people to git clone the repo first? I feel like a few people might come expecting that helm install just magically pulls the chart from 
kubernetes/charts/stable. I know it might feel wrong to be that specific, but can't hurt....

I guess nothing wrong in making it obvious 

> In the README, were you leaving the loadBalancerIP default values blank on purpose? I think the default value for both of them is 'null', isn't it?

I'll add that 
> There are some really subtle grammatical things that I could fix in the morning if you'd like.
feel free :)

> Might it be nice to include a command to check that kong is running. Something like:
 kubectl exec $(kubectl get pods -l app=kong-kong -o jsonpath='{.items[0].metadata.name}') -- curl -s http://kong-kong-admin:8001/status
 But that example still requires people fill in their app name, so it may confuse people as much as it helps. Eh, no big deal if something like this doesn't make it in this round.

Yeah I am not very sure about this, there are multiple other ways to verify that. I did notice that Kong missing `readinessProbe`. I will add that in my free time

Thanks again for all your help

**Comment by shashiranjan84** [comment]
> Do you want to list cassandra.enabled and postgres.enabled, too?
> Since I got tripped up by the multiple flags to switch db's, I'd be happy to add a section explaining that nuance.

 Yes I missed it, feel free to update that.

**Comment by stackedsax** [comment]
@shashiranjan84 you saw my other PR?  Hoping that can be merged into this one.

File: charts/kong/templates/NOTES.txt [hunk 1]
```diff
@@ -1,55 +1,52 @@
...
-     SSL_PORT=$(kubectl get svc --namespace {{ .Release.Namespace }} {{ template "kong.fullname" . }}-admin -o jsonpath='{.spec.ports[1].nodePort}')
 
    {{- else if contains "ClusterIP" .Values.kong.admin.type }}
      HOST=127.0.0.1
 
-     # Execute the following commands to route the connection to Admin port:
-     export POD_NAME=$(kubectl get pods --namespace {{ .Release.Namespace }} -l "app={{ template "kong.fullname" . }}" -o jsonpath="{.items[0].metadata.name}")
-     kubectl port-forward $POD_NAME {{ default "8001" .Values.kong.admin.servicePort }}:{{ default "8001" .Values.kong.admin.servicePort }}
-
+     {{- if .Values.kong.admin.useTLS }}
```
[the remark this answers is not in the document]
Comment (shashiranjan84) [reply charts/kong/templates/NOTES.txt]: For `NodePort` and `LoadBalancer`, PORT will always have the correct port, i.e. `.spec.ports[0].nodePort`

**Comment by shashiranjan84** [comment]
Thanks @stackedsax, its merged. Chart looking way more usable now.

**Pull request merged**